disassemblers/ofrak_cached_disassembly/ofrak_cached_disassembly/components/cached_unpacker.py

```diff
@@ -0,0 +1,305 @@
+
+    def store_program_attributes(self, resource_id: bytes, program_attributes: ProgramAttributes):
+        if resource_id not in self.analysis.keys():
+            self.analysis[resource_id] = dict()
+        self.analysis[resource_id]["program_attributes"] = program_attributes
+
+    def delete_id_from_store(self, resource_id: bytes):
+        if resource_id in self.analysis:
+            del self.analysis[resource_id]
+
+    def get_analysis(self, resource_id: bytes) -> Dict[str, Any]:
+        return self.analysis[resource_id]["analysis"]
+
+    def get_program_attributes(self, resource_id: bytes) -> ProgramAttributes:
+        return self.analysis[resource_id]["program_attributes"]
+
+
+@dataclass
+class CachedAnalysis(ResourceView):
+    pass
+
+
+@dataclass
+class CachedAnalysisAnalyzerConfig(ComponentConfig):
+    filename: str
+    force: Optional[bool] = False
+
+
+class CachedAnalysisAnalyzer(Analyzer[CachedAnalysisAnalyzerConfig, CachedAnalysis]):
+    id = b"CachedAnalysisAnalyzer"
+    targets = (CachedAnalysis,)
+    outputs = (CachedAnalysis,)
+
+    def __init__(
+        self,
+        resource_factory: ResourceFactory,
+        data_service: DataServiceInterface,
+        resource_service: ResourceServiceInterface,
+        analysis_store: CachedAnalysisStore,
+    ):
+        super().__init__(resource_factory, data_service, resource_service)
+        self.analysis_store = analysis_store
+
+    async def analyze(self, resource: Resource, config: CachedAnalysisAnalyzerConfig):
+        await resource.identify()
+        if not resource.has_tag(Program) and not resource.has_attributes(ProgramAttributes):
+            raise AttributeError(
+                f"The reource with ID {resource.get_id()} is not an analyzable program format and does not have ProgramAttributes set."
+            )
+        await resource.unpack()  # Must unpack ELF to get program attributes
+        program_attributes = await resource.analyze(ProgramAttributes)
+        self.analysis_store.store_analysis(resource.get_id(), config.filename)
+        if not config.force:
+            if not await self.verify_cache_file(resource):
+                raise ValueError(
+                    "MD5 recorded in cache file does not match the hash of the requested resource, use the force config option to use this cache file anyway."
+                )
+        self.analysis_store.store_program_attributes(resource.get_id(), program_attributes)
+        cached_analysis_view = CachedAnalysis()
+        resource.add_view(cached_analysis_view)
+        await resource.save()
+        return cached_analysis_view
+
+    async def verify_cache_file(self, resource: Resource):
+        data = await resource.get_data()
+        md5_hash = hashlib.md5(data)
+        return (
+            md5_hash.digest().hex()
+            == self.analysis_store.get_analysis(resource.get_id())["metadata"]["hash"]
+        )
+
+
+class CachedProgramUnpacker(Unpacker[None]):
```

Comment:
This component is unnecessary for the tested usecases and will not run in most cases but could be useful in the event that we are using this to analyze a raw binary.